### ISSUE TITLE
Actual Training Loss Curve

### ISSUE THREAD

**Issue by chenllliang** [FoundationVision/VAR#23]
In the VAR model, a standard cross-entropy loss is employed; Yet Figure 5 illustrates the scaling law on a modified loss function. Could you provide the precise equation detailing how the actual training and test loss are converted to this reduced form? Additionally, is there a training log available that records the actual cross-entropy loss values over the course of the training? Access to this information would greatly enhance readers' comprehension of the training process dynamics for the VAR model. 

Thanks a lot!

**Comment by keyu-tian** [comment]
hi @chenllliang, following previous literature about scaling laws in NLP, we also decompose the standard cross-entropy loss (on the test set) by CE(ground_truth, prediction) = E(ground_truth) + D_KL(ground_truth, prediction). The E(ground_truth) is the so-called *irreducable* loss and D_KL(ground_truth, prediction) the *reducable* loss. Figure 5 illustrates the *reducable* loss, which is calculated by CE - E(ground_truth). E(ground_truth) is calculated on the ground truth token distribution of a trained tokenizer. The E(ground_truth) of the last scale and the average E(ground_truth) are 4.5 and 5, respectively.

I'll post some training logs soon, please stay tuned.

**Comment by chenllliang** [comment]
Thanks for your information!

**Comment by keyu-tian** [comment]
It's obtained by forwarding VQVAE on every ImageNet image, getting the probabilities on each word of the VQ vocabulary, and calculating the entropy.
